apps/ledger-live-mobile/src/hooks/useQuickActions.ts

```diff
@@ -2,102 +2,143 @@ import { useMemo } from "react";
 import { useRoute } from "@react-navigation/native";
 import { useSelector } from "react-redux";
 import { IconsLegacy } from "@ledgerhq/native-ui";
-import { type ParamListBase } from "@react-navigation/native";
 import { IconType } from "@ledgerhq/native-ui/components/Icon/type";
 import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+import { useRampCatalog } from "@ledgerhq/live-common/platform/providers/RampCatalogProvider/useRampCatalog";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
 import { NavigatorName, ScreenName } from "~/const";
+import { BaseNavigatorStackParamList } from "~/components/RootNavigator/types/BaseNavigator";
+import { EntryOf } from "~/types/helpers";
 import { accountsCountSelector, areAccountsEmptySelector } from "../reducers/accounts";
 import { readOnlyModeEnabledSelector } from "../reducers/settings";
 
 export type QuickAction = {
   disabled: boolean;
-  route: [NavigatorName, { screen: ScreenName; params?: ParamListBase }];
+  route: EntryOf<BaseNavigatorStackParamList>;
   icon: IconType;
 };
 
-export type QuickActionsList = {
-  SEND: QuickAction;
-  RECEIVE: QuickAction;
-  BUY: QuickAction;
-  SELL: QuickAction;
-  SWAP: QuickAction;
-  STAKE?: QuickAction;
-  WALLET_CONNECT?: QuickAction;
-  RECOVER?: QuickAction;
-};
+type Actions =
+  | "SEND"
+  | "RECEIVE"
+  | "BUY"
+  | "SELL"
+  | "SWAP"
+  | "STAKE"
+  | "WALLET_CONNECT"
+  | "RECOVER";
+
+type Props = { currency?: CryptoOrTokenCurrency };
 
-function useQuickActions() {
+function useQuickActions({ currency }: Props = {}) {
   const route = useRoute();
 
+  const { isCurrencyAvailable } = useRampCatalog();
+
   const readOnlyModeEnabled = useSelector(readOnlyModeEnabledSelector);
   const accountsCount: number = useSelector(accountsCountSelector);
+  const hasAccounts = accountsCount > 0;
   const areAccountsEmpty = useSelector(areAccountsEmptySelector);
 
   const recoverEntryPoint = useFeature("protectServicesMobile");
   const stakePrograms = useFeature("stakePrograms");
 
   const ptxServiceCtaExchangeDrawer = useFeature("ptxServiceCtaExchangeDrawer");
+  const isPtxServiceCtaExchangeDrawerDisabled = !(ptxServiceCtaExchangeDrawer?.enabled ?? true);
 
-  const isPtxServiceCtaExchangeDrawerDisabled = useMemo(
-    () => !(ptxServiceCtaExchangeDrawer?.enabled ?? true),
-    [ptxServiceCtaExchangeDrawer],
-  );
+  const canBeBought = !currency || isCurrencyAvailable(currency.id, "onRamp");
+  const canBeSold = !currency || currency.id === "bitcoin";
```

Comment:
I'm not sure about this ?

Reply:
Is there a be a better way to detect which currencies can be sold ? I put this check because in the ticket ([LIVE-10741]) it says:
> Please note that Bitcoin is the only coin that has the “Sell” option